migrations/0015_create_messages_table.up.sql

```diff
@@ -0,0 +1,16 @@
+create table messages (
+    id int unsigned not null auto_increment primary key,
+    sender_id int not null,
+    recipient_id int null,
+    recipient_channel varchar(255) null,
+    content varchar(2048) not null,
+    read_at datetime null,
+    created_at datetime not null default CURRENT_TIMESTAMP,
+    deleted_at datetime null
+);
```

Comment:
i know we don't have a practice of this rn, but maybe it's worth a
```sql
CONSTRAINT check_recipient CHECK (recipient_id IS NOT NULL OR recipient_channel IS NOT NULL)
```
? no big deal either way

Reply:
Even better might be 
```sql
CONSTRAINT check_recipient CHECK (
    (recipient_id is null or recipient_channel is null))
    and not (recipient_id is null and recipient_channel is null)
)
```